style: Sync changes from mozilla-central.

emilio opened the pull request:
See each individual commit for details.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/22672)
<!-- Reviewable:end -->

heycam committed:
style: Use atom handles in favour of atom pointers in style system code.

Differential Revision: https://phabricator.services.mozilla.com/D15078

heycam committed:
style: Make GkAtoms opaque to avoid lld-link.exe errors.

Differential Revision: https://phabricator.services.mozilla.com/D15800

emilio committed:
style: Use a less-specific indexmap version.

emilio committed:
style: Rustfmt recent changes.

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!
* These commits modify style code, but no tests are modified. Please consider adding a test!

emilio committed:
style: Three-value position syntax uses calc() as its computed value representation.

This restores the previous behavior of using calc().

Note that background-position / object-position, which test this, weren't
hitting the assertion because they use another codepath.

I didn't add more extensive tests for this because it's well tested for those
two properties, and because this is legacy anyway, see the comment in the test.

I did add the assertion to the codepath those two properties hit.

Differential Revision: https://phabricator.services.mozilla.com/D16176

emilio commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 99f6d6f has been approved by `emilio`

<!-- @bors-servo r=emilio 99f6d6f1b88d5f34c3092546626fa239e75d1c69 -->

bors-servo commented:
:hourglass: Testing commit 99f6d6f1b88d5f34c3092546626fa239e75d1c69 with merge 4b1c5a46373b74e717e0abbebb897fc6a98de492...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/11425)

emilio committed:
Update lockfile.

emilio commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit cc1472e has been approved by `emilio`

<!-- @bors-servo r=emilio cc1472ea2fd79b098d794ee5a73b4d7bbea96e15 -->

bors-servo commented:
:hourglass: Testing commit cc1472ea2fd79b098d794ee5a73b4d7bbea96e15 with merge 3159e11864e4ba01f7646b65cdd21fdb432b3ad5...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/ZQ4c4l7qSeqvsIzng-_3fg)

CYBAI commented:
@bors-servo retry

bors-servo commented:
:bomb: Failed to start rebuilding: `Unknown error`

bors-servo commented:
:hourglass: Testing commit cc1472ea2fd79b098d794ee5a73b4d7bbea96e15 with merge 2a7862540c03e57ff46e8b467f93003940105fb1...

bors-servo commented:
:sunny: Test successful - [android-mac](http://build.servo.org/builders/android-mac/builds/935), [arm32](http://build.servo.org/builders/arm32/builds/13807), [arm64](http://build.servo.org/builders/arm64/builds/13835), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/11427), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/11483), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/6411), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/6379), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/10629), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/10687), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/6437), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/6367), [magicleap](http://build.servo.org/builders/magicleap/builds/554), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/XPQBDwkuT2mx5V9BpN-RTA)
Approved by: emilio
Pushing 2a7862540c03e57ff46e8b467f93003940105fb1 to master...
